Libraries/xcsdk/Tools/xcrun.cpp

```diff
@@ -298,17 +298,70 @@ main(int argc, char **argv)
     /*
      * Determine the SDK to use.
      */
-    xcsdk::SDK::Target::shared_ptr target = manager->findTarget(SDK.value_or("macosx"));
-    if (target == nullptr) {
-        if (SDK) {
-            fprintf(stderr, "error: unable to find sdk '%s'\n", SDK->c_str());
+    const std::string defaultSDK = "macosx";
+    xcsdk::SDK::Target::shared_ptr target = manager->findTarget(SDK.value_or(defaultSDK));
```

Comment:
I think it should be OK to not specify any `-sdk` and have no default, but if you do specify an `-sdk`, not finding that SDK should be an error. Sort of like this:

    xcsdk::SDK::Target::shared_ptr target = nullptr;
    if (SDK) {
        target = manager->findTarget(*SDK);
        if (target == nullptr) {
             /* error */
        }
    } else {
        target = target = manager->findTarget(defaultSDK);
        /* nullptr target is not an error */
    }